src/platform/telink/BUILD.gn

```diff
@@ -62,5 +62,7 @@ static_library("telink") {
     }
   }
 
-  public_deps += [ "${chip_root}/src/crypto" ]
+  public_deps += [ 
+    "${chip_root}/src/crypto",
+    "${chip_root}/src/setup_payload" ]
```

Comment:
same comment here - depending inside platform on a larger library seems a potential problem. Lets split it into  `${chip_root}/src/setup_payload:data_types` or similar for just minimal structures and constants.

Reply:
It sounds reasonable, I extracted additional data files only to separate module, as only those one are needed in this particular case. Please check if this address your suggestion.